core/src/Kokkos_NumericTraits.hpp

```diff
@@ -306,6 +306,30 @@ struct reduction_identity<signed char> {
   }
 };
 
+template <>
+struct reduction_identity<bool> {
+  KOKKOS_FORCEINLINE_FUNCTION constexpr static bool sum() {
```

Comment:
should we not implement sum/prod/max/min? It feels like those aren't appropriate for bool

Reply:
ah I just saw that the tests were also ommitted.

Reply:
Thank you.  Yeah... I wasn't sure about those either.  2e4d16a removes them.